wlauto/workloads/googlephotos/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -145,55 +137,260 @@ private void gesturesTest() throws Exception {
+
+        private Position seekBarPosition;
+        private int percent;
+        private int steps;
+
+        SeekBarTestParams(final Position position, final int steps, final int percent) {
+            this.seekBarPosition = position;
+            this.steps = steps;
+            this.percent = percent;
+        }
+    }
+
+    private void editPhotoColorTest() throws Exception {
         String testTag = "edit_photo";
 
-        Timer result = new Timer();
-        result.start();
+        // Perform a range of swipe tests while browsing photo gallery
+        LinkedHashMap<String, SeekBarTestParams> testParams = new LinkedHashMap<String, SeekBarTestParams>();
+        testParams.put("increment_color", new SeekBarTestParams(Position.RIGHT, 10, 20));
+        testParams.put("reset_color", new SeekBarTestParams(Position.CENTRE, 0, 0));
+        testParams.put("decrement_color", new SeekBarTestParams(Position.LEFT, 10, 20));
 
-        // Select first photograph
-        selectPhoto(0);
+        Iterator<Entry<String, SeekBarTestParams>> it = testParams.entrySet().iterator();
+
+        // Select second photograph
+        selectPhoto(1);
         UiObject editView = getUiObjectByResourceId("com.google.android.apps.photos:id/edit",
                                                     "android.widget.ImageView");
         editView.click();
 
-        UiObject editColor = getUiObjectByText("Colour", "android.widget.RadioButton");
-        editColor.click();
+        // Manage potential different spelling of UI element
+        UiObject editColor = new UiObject(new UiSelector().text("Color"));
+        UiObject editColour = new UiObject(new UiSelector().text("Colour"));
+
+        if (editColor.exists()) {
+            editColor.click();
+        } else if (editColour.exists()) {
+            editColour.click();
+        } else {
+            throw new UiObjectNotFoundException(String.format("Could not find \"%s\" \"%s\"",
+                                                              "Color/Colour", "android.widget.RadioButton"));
+        }
 
         UiObject seekBar = getUiObjectByResourceId("com.google.android.apps.photos:id/cpe_strength_seek_bar",
                                                    "android.widget.SeekBar");
-        seekBar.swipeLeft(10);
 
-        UiObject accept = getUiObjectByDescription("Accept", "android.widget.ImageView");
-        accept.click();
+        String viewName = "com.google.android.apps.consumerphotoeditor.fragments.ConsumerPhotoEditorActivity";
 
-        UiObject save = getUiObjectByText("SAVE", "android.widget.TextView");
-        save.click();
+        while (it.hasNext()) {
+            Map.Entry<String, SeekBarTestParams> pair = it.next();
+            Position pos = pair.getValue().seekBarPosition;
+            int steps = pair.getValue().steps;
+            int percent = pair.getValue().percent;
 
-        // Return to application home screen
-        getUiDevice().pressBack();
```

Comment:
Although a simpler approach this method wasn't as reliable for me as finding a UiObject clicking.